static/scripts/newblocks.js

```diff
@@ -373,7 +371,23 @@ Blockly.JavaScript["ea_addtopopulation"] = function (block) {
       length +
       ")";
   } else if (dropdown_selection_strategy == "CHANCE") {
-    code += "shuffle(parent_population).slice(0," + length + ")";
+    //global shuffle function from https://stackoverflow.com/a/2450976 
+    var functionName = Blockly.JavaScript.provideFunction_("shuffle", [
+      "function " +
+      Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ +
+      "(array) {",
+      " var currentIndex = array.length, temporaryValue, randomIndex;",
+      " while (0 !== currentIndex) {",
+      "   randomIndex = Math.floor(Math.random() * randomIndex);",
+      "   currentIndex -= 1;",
+      "   temporaryValue = array[currentIndex];",
+      "   array[currentIndex] = array[randomIndex];",
+      "   array[randomIndex] = temporaryValue;",
+      " }",
+      " return array;",
+      "}",
+    ])
+    code += functionName + POPULATION + ".slice(0," + length + ")";
```

Comment:
I think brackets to the function call are still missing here

Reply:
Yes. They are provided by the `POPULATION` block. 
Since we probably want to change that later on we will go with double brackets.